### PR TITLE
Fix/condo/doma 2567/fix incidents after testing

### PR DESCRIPTION
**Pull request by Alllex202**
for https://github.com/open-condo-software/condo/pull/2821

### CODE REVIEW

File: apps/condo/domains/ticket/components/BaseTicketForm/index.tsx [hunk 1]
```diff
@@ -352,7 +352,7 @@ export const TicketSourceSelect: React.FC = () => {
 
 const FORM_VALIDATE_TRIGGER = ['onBlur', 'onSubmit']
 const TICKET_PROPERTY_HINT_STYLES: CSSProperties = { maxHeight: '11em', maxWidth: '250px' }
-const HINTS_WRAPPER_STYLE: CSSProperties = { overflow: 'auto', maxHeight: 'calc(100vh - 128px)', paddingRight: 8 }
+const HINTS_WRAPPER_STYLE: CSSProperties = { overflowY: 'scroll', maxHeight: 'calc(100vh - 220px)', paddingRight: 8 }
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/ticket/components/BaseTicketForm/index.tsx:355]: What does the scrollbar look like in chrome? 
In callcenter together with the organization plate?

File: apps/condo/domains/ticket/components/IncidentHints.tsx [hunk 2]
```diff
@@ -196,7 +202,7 @@ export const IncidentHints: React.FC<IncidentHintsProps> = (props) => {
         )
         const notActualLastIncidents = await fetchIncidents(
             [SortIncidentsBy.WorkFinishDesc, SortIncidentsBy.CreatedAtDesc],
-            incidentIds, organizationId, IncidentStatusType.NotActual, 3
+            incidentIds, organizationId, IncidentStatusType.NotActual, 7
         )
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/ticket/components/IncidentHints.tsx:206]: Maybe better to use one parameter - object.
For better readability

File: apps/condo/domains/ticket/utils/clientSchema/IncidentRenders.tsx [hunk 3]
```diff
@@ -134,8 +136,9 @@ export const getManyIncidentClassifiersGroupByPlaceRender = (ellipsis?: boolean
         const problemsPart = !isEmpty(problemNames) ? ` » ${problemNames.join(', ')}` : ''
 
         const text = `${categoriesPart}${problemsPart}`
+        const trimmedText = text.length > MAX_CELL_CONTENT_LENGTH ? `${text.substring(0, MAX_CELL_CONTENT_LENGTH)}…` : text
 
-        return getTableCellRenderer(null, ellipsis)(text)
+        return getTableCellRenderer(null, false, null, null, null, text)(trimmedText)
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/ticket/utils/clientSchema/IncidentRenders.tsx:141]: I think here we also need to use the object parameter

**Review by nomerdvadcatpyat — APPROVED** [review]

File: apps/condo/domains/ticket/utils/clientSchema/IncidentRenders.tsx (hunk 3 above)
Comment (Alllex202) [reply apps/condo/domains/ticket/utils/clientSchema/IncidentRenders.tsx:141]: I'm afraid to refactor it))

Comment (Alllex202) [reply apps/condo/domains/ticket/utils/clientSchema/IncidentRenders.tsx:141]: It is worth starting a new task for this

File: apps/condo/domains/ticket/components/IncidentHints.tsx (hunk 2 above)
Comment (Alllex202) [reply apps/condo/domains/ticket/components/IncidentHints.tsx]: fixed it

File: apps/condo/domains/ticket/components/IncidentHints.tsx [hunk 4]
```diff
@@ -46,6 +46,12 @@ const IncidentHint: React.FC<IncidentHintProps> = (props) => {
 
     const { incident } = props
 
+    const trimmedDetails = useMemo(() => {
+        const details = get(incident, 'details', '')
+        return details.length > MAX_DETAILS_LENGTH ? `${details.substring(0, MAX_DETAILS_LENGTH)}…` : details
+    }, [incident])
```
Comment (akarjakin) [inline apps/condo/domains/ticket/components/IncidentHints.tsx:52]: this code depends on `details.length` too

File: apps/condo/domains/ticket/components/IncidentForm/BaseIncidentForm.tsx [hunk 5]
```diff
@@ -417,11 +404,23 @@ export const BaseIncidentForm: React.FC<BaseIncidentFormProps> = (props) => {
...
         ...initialValues,
         properties: initialPropertyIds,
         classifiers: initialIncidentClassifiers,
     }), [initialIncidentClassifiers, initialPropertyIds, initialValues])
 
+    const workTypeLabels = useMemo(() => ({
+        [INCIDENT_WORK_TYPE_SCHEDULED]: ScheduledTypeLabel,
+        [INCIDENT_WORK_TYPE_EMERGENCY]: EmergencyTypeLabel,
+    }), [EmergencyTypeLabel, ScheduledTypeLabel])
+    const workTypeOptions = useMemo(() => INCIDENT_WORK_TYPES.map((type) => {
```
Comment (akarjakin) [inline apps/condo/domains/ticket/components/IncidentForm/BaseIncidentForm.tsx:417]: code style like this is hard to read and understand.

File: apps/condo/domains/ticket/components/IncidentHints.tsx (hunk 4 above)
Comment (akarjakin) [reply apps/condo/domains/ticket/components/IncidentHints.tsx:52]: Trimming text by length does not take into account block width, font size, etc. 
Are you sure it's the best solution?

File: apps/condo/domains/ticket/components/IncidentHints.tsx [hunk 6]
```diff
@@ -190,14 +200,19 @@ export const IncidentHints: React.FC<IncidentHintsProps> = (props) => {
...
+            incidentIds,
+            organizationId,
+            status: IncidentStatusType.Actual,
+        })
+        const notActualLastIncidents = await fetchIncidents({
+            sortBy: [SortIncidentsBy.WorkFinishDesc, SortIncidentsBy.CreatedAtDesc],
+            incidentIds,
+            organizationId,
+            status: IncidentStatusType.NotActual,
+            lastDays: 7,
```
Comment (akarjakin) [inline apps/condo/domains/ticket/components/IncidentHints.tsx:214]: constants should have proper naming.

File: apps/condo/domains/ticket/hooks/useIncidentUpdateStatusModal.tsx [hunk 7]
```diff
@@ -69,13 +69,19 @@ export const useIncidentUpdateStatusModal: UseIncidentUpdateStatusModalType = ({
...
     const initialState = useMemo(() => {
         const beforeWorkStart = incident.workStart && dayjs().diff(incident.workStart) < 0
-        const workFinish = incident.workFinish
-            ? dayjs(incident.workFinish)
-            : beforeWorkStart
-                ? dayjs(incident.workStart)
-                : dayjs()
+        const afterWorkFinish = incident.workFinish && dayjs().diff(incident.workFinish) > 0
+
+        let workFinish
```
Comment (akarjakin) [inline apps/condo/domains/ticket/hooks/useIncidentUpdateStatusModal.tsx:74]:
```
let someDate //please fix this naming according to meaning

if (!incident.workFinish && beforeWorkStart) someDate = incident.workStart
if (!someDate && isActual && afterWorkFinish) someDate = incident.workFinish

const workFinish = dayjs(someDate)
``

Comment (akarjakin) [reply apps/condo/domains/ticket/hooks/useIncidentUpdateStatusModal.tsx:74]: this way code will be a bit more readable

File: apps/condo/domains/ticket/schema/ExportIncidentsToExcelService.js [hunk 8]
```diff
@@ -53,13 +55,11 @@ const classifiersToString = (classifiers = []) => {
...
 
-    return `${place} » ${categories}${renderProblems}`
+    return `${categories}${renderProblems}`
 }
 
-const formatDate = (date, timeZone, format = 'DD.MM.YYYY HH:mm') => {
-    if (!date || !format) return EMPTY_SYMBOL
-    if (!timeZone) return dayjs(date).format(format)
-    return dayjs(date).tz(timeZone).format(format)
+const formatDate = (date, timeZone, format) => {
```
Comment (akarjakin) [inline apps/condo/domains/ticket/schema/ExportIncidentsToExcelService.js:61]: please use default value like `format = DATE_FORMAT`

File: apps/condo/migrations/20230221010326-0223_incident_incidentclassifier_and_more.js [hunk 9]
```diff
@@ -14,7 +14,7 @@ exports.up = async (knex) => {
...
+CREATE TABLE "Incident" ("number" integer NOT NULL UNIQUE, "details" text NOT NULL, "status" varchar(50) NOT NULL, "textForResident" text NULL, "workStart" timestamp with time zone NOT NULL, "workFinish" timestamp with time zone NULL, "workType" varchar(50) NULL, "hasAllProperties" boolean NOT NULL, "id" uuid NOT NULL PRIMARY KEY, "v" integer NOT NULL, "createdAt" timestamp with time zone NULL, "updatedAt" timestamp with time zone NULL, "deletedAt" timestamp with time zone NULL, "newId" uuid NULL, "dv" integer NOT NULL, "sender" jsonb NOT NULL, "createdBy" uuid NULL, "organization" uuid NOT NULL, "updatedBy" uuid NULL);
```
Comment (akarjakin) [inline apps/condo/migrations/20230221010326-0223_incident_incidentclassifier_and_more.js:17]: you have lots of potentially searchable fields here like `status`, `number`, `workStart`, `workFinish`, etc. but no indexes added, that will make such searching very slow.

File: apps/condo/domains/ticket/components/IncidentHints.tsx (hunk 4 above)
Comment (Alllex202) [reply apps/condo/domains/ticket/components/IncidentHints.tsx:52]: This is the designers decision, not mine

code depends on `incident` only

File: apps/condo/domains/ticket/schema/ExportIncidentsToExcelService.js (hunk 8 above)
Comment (Alllex202) [reply apps/condo/domains/ticket/schema/ExportIncidentsToExcelService.js:61]: Does it make sense? It is below overridden if empty

File: apps/condo/migrations/20230221010326-0223_incident_incidentclassifier_and_more.js (hunk 9 above)
Comment (Alllex202) [reply apps/condo/migrations/20230221010326-0223_incident_incidentclassifier_and_more.js:17]: If it's really needed, we'll add it later

File: apps/condo/domains/common/components/containers/GlobalStyle.tsx [hunk 10]
```diff
@@ -322,6 +322,10 @@ export default function GlobalStyle () {
                   }
                 }
               }
+              
+              .ant-picker-input > input {
```
Comment (pahaz) [inline apps/condo/domains/common/components/containers/GlobalStyle.tsx:326]: should it be a part of a component?

Comment (Alllex202) [reply apps/condo/domains/common/components/containers/GlobalStyle.tsx]: moved it to  `Picker` component

**Pull request merged**